src/scripts/main.js

```diff
@@ -1,3 +1,21 @@
 'use strict';
 
-// write your code here
+function addCellWithFirstColumnText(selector) {
+  const rows = document.querySelectorAll(selector);
+
+  rows.forEach((row) => {
+    const firstCellText = row.cells[1].textContent;
+
+    const isHeaderRow =
+      row.parentElement.tagName === 'THEAD' ||
+      row.parentElement.tagName === 'TFOOT';
+    const newCell = document.createElement(isHeaderRow ? 'th' : 'td');
+
+    newCell.textContent = firstCellText;
+    row.insertBefore(newCell, row.cells[4]);
```

Comment:
Inserting the new cell before `row.cells[4]` assumes that there are at least five cells in each row. This may not be the case, and could lead to errors if a row has fewer than five cells. Consider checking the number of cells or inserting the new cell at a different position.